File: client/lib/features/settings/data/models/saved_collection_model.dart
class SavedCollectionModel {
  final String id;
  final String userId;
  final String name;
  final String? coverPostId;
  final int postCount;
  final bool isDefault;

  SavedCollectionModel({
    required this.id,
    required this.userId,
    required this.name,
    this.coverPostId,
    required this.postCount,
    required this.isDefault,
  });

  factory SavedCollectionModel.fromJson(Map<String, dynamic> json) {
    return SavedCollectionModel(
      id: json['id'] ?? '',
      userId: json['userId'] ?? json['user_id'] ?? '',
      name: json['name'] ?? '',
      coverPostId: json['coverPostId'] ?? json['cover_post_id'],
      postCount: json['postCount'] ?? json['post_count'] ?? 0,
      isDefault: json['isDefault'] ?? json['is_default'] ?? false,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'userId': userId,
    'name': name,
    'coverPostId': coverPostId,
    'postCount': postCount,
    'isDefault': isDefault,
  };
}

File: server/src/models/SavedCollection.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const SavedCollection = sequelize.define(
  'SavedCollection',
  {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
      allowNull: false,
    },
    userId: {
      type: DataTypes.UUID,
      allowNull: false,
      field: 'user_id',
      references: { model: 'users', key: 'id' },
      onDelete: 'CASCADE',
    },
    name: {
      type: DataTypes.STRING(60),
      allowNull: false,
      validate: {
        len: [1, 60],
      },
    },
    coverPostId: {
      type: DataTypes.UUID,
      allowNull: true,
      field: 'cover_post_id',
      references: { model: 'posts', key: 'id' },
      onDelete: 'SET NULL',
    },
    postCount: {
      type: DataTypes.INTEGER,
      defaultValue: 0,
      field: 'post_count',
    },
    isDefault: {
      type: DataTypes.BOOLEAN,
      defaultValue: false,
      field: 'is_default',
    },
  },
  {
    tableName: 'saved_collections',
    timestamps: true,
    underscored: true,
  }
);

module.exports = SavedCollection;
